compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -2144,6 +2152,13 @@ where
             | ty::Param(_)
             | ty::Infer(_)
             | ty::Error(_) => bug!("TyAndLayout::field_type: unexpected type `{}`", this.ty),
+        }
+    }
+
+    fn field(this: TyAndLayout<'tcx>, cx: &C, i: usize) -> C::TyAndLayout {
```

Comment:
what is the motivation to extract this helper function?

Reply:
we can't recurse on a match block as far as I know

Reply:
Essentially the call is 
```
cx.layout_of(match {

})
```

One way would be to construct a new `TyAndLayout` with the tuple instead of Closure as `this.ty` but I'm not sure if anything else needs to change around it